atecc608a_se.c

```diff
@@ -386,22 +393,35 @@ psa_status_t atecc608a_read(uint16_t slot, size_t offset, uint8_t *data, size_t
     return status;
 }
 
-psa_status_t atecc608a_check_slot(psa_drv_se_context_t *drv_context,
-                                  const psa_key_attributes_t *attributes,
-                                  psa_key_slot_number_t key_slot)
+static psa_status_t atecc608a_validate_slot_number(
+    psa_drv_se_context_t *drv_context,
+    const psa_key_attributes_t *attributes,
+    psa_key_creation_method_t method,
+    psa_key_slot_number_t key_slot)
 {
-    if (PSA_KEY_TYPE_IS_ECC_KEY_PAIR(attributes->type)) {
+    psa_key_type_t type = psa_get_key_type(attributes);
+    if (PSA_KEY_TYPE_IS_ECC_KEY_PAIR(type)) {
         if (key_slot <= 15) {
             return PSA_SUCCESS;
         }
-    } else if (PSA_KEY_TYPE_IS_ECC_PUBLIC_KEY(attributes->type)) {
+    } else if (PSA_KEY_TYPE_IS_ECC_PUBLIC_KEY(type)) {
         if (key_slot >= 8 && key_slot <= 15) {
             return PSA_SUCCESS;
         }
     }
     return PSA_ERROR_NOT_SUPPORTED;
 }
 
+static psa_status_t atecc608a_allocate_key(
+    psa_drv_se_context_t *drv_context,
+    void *persistent_data,
+    const psa_key_attributes_t *attributes,
+    psa_key_creation_method_t method,
+    psa_key_slot_number_t *key_slot)
+{
+    return PSA_SUCCESS;
```

Comment:
This should choose a key_slot according to the attributes, but for now can be a dummy.